virtwho/virt/xen/xen.py

```diff
@@ -121,14 +168,14 @@ def getHostGuestMapping(self):
             if version:
                 facts[virt.Hypervisor.HYPERVISOR_VERSION_FACT] = version
 
-            if self.config.hypervisor_id == 'uuid':
+            if self.config['hypervisor_id'] == 'uuid':
```

Comment:
Should this be moved to a _validate_hypervisor_id method?

Would return an error message if the value provided was not one of those expected and valid for Xen.

Reply:
You are right. The `VirtConfigSection` has only lambda function returning `None` in all cases. I will fix it for other virt. backends (hyper-v and libvirtd) in this PR or I can create special PR when Xen and RHEVM fill be merged to config_refactor.

Reply:
Solved in PR #105. When both PRs will be merged to config_refactor branch, then it should work as expected.